### DISCUSSION TITLE
순서보장에 관한 질문들

### DISCUSSION THREAD

**Discussion by sejongjeong** [kyungjunleeme/kafka_study#25] [Q&A]
enable.idempotence 옵션을 키면 어떻게 순서 보장이 되는건지? (멱등성은 중복이 없음을 보장하는 개념이 아닌지)

key 값이 동일할때 브로커내 한 파티션에서 순서 보장이 된다고 했는데 producer는 비동기적으로 동작하므로 thread 1 이 정보를 보내다가 에러가 나서 재처리를 위해 블록 되었을때 thread 2가 send 한 정보가 먼저 도착하면 어떻게 브로커내 파티션에서 순서가 보장되는지

**Comment by kyungjunleeme** [comment]
`acks = all` 이 활성화 된다는 내용도 알아야함
